byte-buddy-dep/src/test/java/net/bytebuddy/implementation/auxiliary/AbstractMethodCallProxyTest.java

```diff
@@ -54,13 +55,26 @@ protected Class<?> proxyOnlyDeclaredMethodOf(Class<?> proxyTarget) throws Except
         assertThat(Runnable.class.isAssignableFrom(auxiliaryType), is(true));
         assertThat(auxiliaryType.getDeclaredConstructors().length, is(1));
         assertThat(auxiliaryType.getDeclaredMethods().length, is(2));
-        assertThat(auxiliaryType.getDeclaredFields().length, is(proxyMethod.getParameters().size() + (proxyMethod.isStatic() ? 0 : 1)));
-        int fieldIndex = 0;
-        if (!proxyMethod.isStatic()) {
-            assertThat(auxiliaryType.getDeclaredFields()[fieldIndex++].getType(), CoreMatchers.<Class<?>>is(proxyTarget));
+        assertThat(auxiliaryType.getDeclaredFields().length, is(proxyMethod.getParameters().size() + (proxyMethod.isStatic() ? 0 : 1)));       
+        Field targetField = null;
+        for (Field field : auxiliaryType.getDeclaredFields()) {
+            if (field.getType() == proxyTarget) {
+                targetField = field;
+                break;
+            }
         }
-        for (Class<?> parameterType : proxyTarget.getDeclaredMethods()[0].getParameterTypes()) {
-            assertThat(auxiliaryType.getDeclaredFields()[fieldIndex++].getType(), CoreMatchers.<Class<?>>is(parameterType));
+        if (!proxyMethod.isStatic() && targetField != null) {
+            assertThat(targetField.getType(), CoreMatchers.<Class<?>>is(proxyTarget));
+        }
+        for (Class<?> parameterType : proxyTarget.getDeclaredMethods()[0].getParameterTypes()){
```

Comment:
This for loop doesn't seem to be doing what L62 is doing.

L62 iterates over each parameter type and checks whether `auxiliaryType.getDeclaredFields()[fieldIndex++].getType()` in this order matches the expected parameter types.

This for loop here is just checking whether the parameter types exist in `auxiliaryType.getDeclaredFields()[fieldIndex++].getType()`, but it (1) overlooks the ordering and (2) overlooks the number of times in which a parameter type should exist.

For (1), it is likely this ordering that is causing the test to be flaky so it is fine to ignore. But for (2), a better fix should still properly check the count of each parameter type.